src/loki/worker.cc

```diff
@@ -71,6 +71,16 @@ void loki_worker_t::parse_costing(Api& api, bool allow_none) {
     throw valhalla_exception_t{124};
   }
 
+  auto costings = options.costings();
+  for (const auto& pair : costings) {
+    if (!allow_hard_exclusions &&
+        (pair.second.options().exclude_bridges() || pair.second.options().exclude_tolls() ||
+         pair.second.options().exclude_tunnels() || pair.second.options().exclude_highways() ||
+         pair.second.options().exclude_ferries())) {
+      throw valhalla_exception_t{145};
```

Comment:
make this a warning please, not an exception. there's nothing a user can do about it anyways, so he's forced to send the same request again, might as well let it pass right away instead of throwing.

Reply:
In this [comment](https://github.com/valhalla/valhalla/pull/3733#issuecomment-1235883038 ) if it is detected that one of these options is set and its not allowed, return a 400

Reply:
i agree with nils, a warning and just setting the option back to false is far better than a 400. the user still gets a usable response but they are told that we couldnt adhere to what they wanted so we gave them the next best thing

Reply:
Done